### DISCUSSION TITLE
How much mobile data does this app use? Can I pre-download assets for my region?

### DISCUSSION THREAD

**Discussion by Swarkin** [streetcomplete/StreetComplete#5652] [Q&A] [answered]
I'm very limited by data (less than a GB) and would like to know whether this app has some low-data mode or even fully-offline mode while I'm outside?

**Comment by FloEdelmann** [comment, marked as answer]
You can turn off automatic uploads in the app settings and download the data manually for a specific region via the ≡ button (zoom level must be high enough, though, so you might have to do this multiple times). Then you can use the app completely offline.

**Reply by Swarkin** [reply]
Is it possible to download data for my entire city? Scrolling around and pressing download is a bit annoying

**Reply by matkoniecz** [reply]
> Is it possible to download data for my entire city? 

via manual downloads

though typically just small part of city will provide quests for multihour walk

> Scrolling around and pressing download is a bit annoying

you can zoom out just a bit before download is no longer possible (note that weaker devices + dense area can result in really long processing)

> I'm very limited by data (less than a GB) 

Note that app will refetch data if it is old. But it will work fine in offline mode if data was preloaded

(few years ago I was in such situation and SC worked fine, ot should still work)

> You can turn off automatic uploads in the app settings

Remember to make manual uploads in such case!

**Reply by FloEdelmann** [reply]
> Is it possible to download data for my entire city?

Yes, but not at once; only with scrolling around and pressing download :wink: 

> is a bit annoying

Yes, but the download area is limited to not overload the server and the phone which needs to process all the data.

There is a feature request to download along a GPX track (#473, #1757) that would mitigate that, and also a (abandoned?) work-in-progress PR (#5369). It sequentially downloads multiple areas along the track.

**Reply by Swarkin** [reply]
Thanks for the info all!

**Reply by mnalis** [reply]
> There is a feature request to download along a GPX track (https://github.com/streetcomplete/StreetComplete/issues/473, https://github.com/streetcomplete/StreetComplete/issues/1757) that would mitigate that, and also a (abandoned?) work-in-progress PR (https://github.com/streetcomplete/StreetComplete/pull/5369). It sequentially downloads multiple areas along the track.

Relatedly, [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) (StreetComplete _"Expert Edition"_ fork) does already implement downloads along the GPX tracks. It also allows to disable automatic downloads (in case you might prefer that to completely disabling your mobile data for all apps)

**Comment by timothywashere** [comment]
I did some experiments with Street complete. If you pre-download the data it uses very little when you are walking around. See the last post get  #3556
